source/routing/raptor_api.cpp

```diff
@@ -1368,73 +1369,79 @@ void filter_late_journeys(RAPTOR::Journeys& journeys, const NightBusFilter::Para
     }
 }
 
-bool find_stop_area_of_interest_through_journey(const navitia::routing::Journey& journey,
-                                                const std::string& stop_area_uri,
-                                                const navitia::type::VehicleJourney* vj_to_skip) {
-    bool found = false;
-
-    for (const auto& section : journey.sections) {
-        auto order = section.get_in_st->order();
-        bool reach_end = false;
-        // because of stay-ins, we may have several vj in one section, we have to scan the stop times
-        // of all vjs
-        for (const auto* vj = section.get_in_st->vehicle_journey; vj;
-             (vj = vj->next_vj, order = type::RankStopTime(0))) {
-            for (const auto& st :
-                 boost::make_iterator_range(vj->stop_time_list.begin() + order.val, vj->stop_time_list.end())) {
-                if (st.stop_point->stop_area->uri == stop_area_uri && vj != vj_to_skip) {
-                    return true;
-                }
-                if (section.get_out_st == &st) {
-                    reach_end = true;
-                    break;
+bool update_visited_section(navitia::routing::Journey::Section& section,
+                            const std::string& stop_area_uri,
+                            const navitia::type::VehicleJourney* vj_to_skip,
+                            const bool clockwise) {
+    log4cplus::Logger logger = log4cplus::Logger::getInstance(LOG4CPLUS_TEXT("logger"));
+    auto order = section.get_in_st->order();
+    // because of stay-ins, we may have several vj in one section, we have to scan the stop times
+    // of all vjs
+    for (const auto* vj = section.get_in_st->vehicle_journey; vj; (vj = vj->next_vj, order = type::RankStopTime(0))) {
+        for (const auto& st :
+             boost::make_iterator_range(vj->stop_time_list.begin() + order.val, vj->stop_time_list.end())) {
+            if (st.stop_point->stop_area->uri == stop_area_uri && vj != vj_to_skip) {
+                if (clockwise) {
+                    section.get_out_st = &st;
+                    section.get_out_dt = st.departure_time;
```

Comment:
```suggestion
                    section.get_out_dt = st.arrival_time;
```
you "get out" at "arrival_time"